src/core/datagram.c

```diff
@@ -583,3 +575,19 @@ QuicDatagramProcessFrame(
 
     return TRUE;
 }
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicDatagramCancelPending(
+    _In_ QUIC_CONNECTION* Connection
+    )
+{
+    QUIC_DATAGRAM* Datagram = &Connection->Datagram;
+    while (Datagram->SendQueue != NULL) {
+        QUIC_SEND_REQUEST* SendRequest = Datagram->SendQueue;
+        Datagram->SendQueue = SendRequest->Next;
+        QuicDatagramCancelSend(Connection, SendRequest);
```

Comment:
Only cancel sends that have the appropriate flag. It's possible some do and some don't.

Reply:
I might need some review on this one lol